### ISSUE TITLE
Ensure User-Agent is sent

### ISSUE THREAD

**Issue by nbarbettini** [okta/okta-sdk-appauth-android#67]
Verify that this SDK sends a User-Agent header whenever it:

* Retrieves the OpenID Connect metadata document (`/.well-known/openid-configuration`)
* Exchanges an authorization code for tokens, for *both* browser or native sign-in

The value of the header must be:

```
User-Agent: okta-sdk-appauth-android/{version} {osVersion} {deviceInfo}
```

The SDK may already be doing this in all cases, but I want to double check 🙂 

More info can be found here: https://oktawiki.atlassian.net/wiki/spaces/PM/pages/237672215/User-Agent+and+X-Okta-User-Agent-Extended

**Comment by bdemers** [comment]
When usage of the management or authn sdk, just create a `META-INF/okta/version.properties` file with the format of:
```
okta-sdk-appauth-android=${version-at-build-time}
```

https://github.com/okta/okta-jwt-verifier-java/blob/master/impl/src/main/resources/META-INF/okta/version.properties

For appAuth usage, the easiest way to implement the same format is to depend on `com.okta.commons:okta-commons-lang` and use [`ApplicationInfo`](https://github.com/okta/okta-commons-java/blob/master/lang/src/main/java/com/okta/commons/lang/ApplicationInfo.java)

Though all of these options depend on Java8 functionality, so you might need to port this logic , maybe just this method:?
https://github.com/okta/okta-commons-java/blob/master/lang/src/main/java/com/okta/commons/lang/ApplicationInfo.java#L311-L322

**Comment by smokienko** [comment]
@nbarbettini Currently our SDK sends User-Agent token only as a part of Authorized requests. 
In order to change "User-Agent" header for token exchange and Open Id Config request, we need to override those requests within AppAuth code. We can do it as a part of moving from AppAuth SDK.

**Comment by nbarbettini** [comment]
@smokienko Makes sense. Bubbling this up again so we don't lose it when talking about migrating from AppAuth.
